watchtower-plugin/src/net/http.rs

```diff
@@ -119,21 +119,25 @@ pub async fn send_appointment(
 
 /// Generic function to post different types of requests to the tower.
 pub async fn post_request<S: Serialize>(endpoint: &str, data: S) -> Result<Response, RequestError> {
-    reqwest::Client::new()
-        .post(endpoint)
-        .json(&data)
-        .send()
-        .await
-        .map_err(|e| {
-            log::error!("{}", e);
-            if e.is_connect() | e.is_timeout() {
-                RequestError::ConnectionError(
-                    "Cannot connect to the tower. Connection refused".into(),
-                )
-            } else {
-                RequestError::Unexpected("Unexpected error ocurred (see logs for more info)".into())
-            }
-        })
+    let client = if endpoint.contains(".onion:") {
```

Comment:
This will fail if the tower's onion service is running on 443 (https) or 80 (http) and the port is not specified.
we can do something along the lines of:
```rust
let url = reqwest::Url::from(endpoint);
if url.host().unwrap().to_string().ends_with(".onion") {
    // some stuff
}
```

Reply:
Fixed in [dcbcee3](https://github.com/talaia-labs/rust-teos/pull/96/commits/dcbcee385a45aaf36abf1d823fa6e79ae941325b)